aruco_ros/src/aruco_detector.cpp

```diff
@@ -189,14 +228,65 @@ class ArucoSimple
         //Ok, let's detect
         mDetector.detect(inImage, markers, camParam, marker_size, false);
         //for each marker, draw info and its boundaries in the image
+
+        bool multiple_markers_detected = markers.size() > 1;
+        unsigned int error_condition = 0;
+        std::string error_message = "";
+
         for(size_t i=0; i<markers.size(); ++i)
         {
 
-          if (markers.size() == 1){  // Only process markers if there is only one in FOV
+          if (markers.size() == 1){
```

Comment:
This logic looks so ugly, why not check the size before the loop? It would make it much more readable, I think, and put all the logic for single marker in a separate function